sql/core/src/test/scala/org/apache/spark/sql/connector/KeyGroupedPartitioningSuite.scala

```diff
@@ -310,9 +310,8 @@ class KeyGroupedPartitioningSuite extends DistributionAndOrderingSuiteBase {
 
   test("partitioned join: only one side reports partitioning") {
     val customers_partitions = Array(bucket(4, "customer_id"))
-    val orders_partitions = Array(bucket(2, "customer_id"))
 
-    testWithCustomersAndOrders(customers_partitions, orders_partitions, 2)
```

Comment:
It seems `testWithCustomersAndOrders(customers_partitions, orders_partitions, 2)` is OK too ?

Reply:
Shall we modify testWithCustomersAndOrders and let the expected results are different ? Otherwise, The test cases looks confused.

Reply:
why the test didn't fail before your PR?